### ISSUE TITLE
openshift console to have a link/menu to other apps/consoles?

### ISSUE THREAD

**Issue by jstrachan** [openshift/origin#1977]
there's going to be a number of web apps/consoles that folks might be running inside a project/namespace. 

e.g. today off the top of my head the consoles I tend to use on OpenShift are: kibana, grafana, swagger, ApiMan, Jenkins, Nexus and fabric8's API Registry console to name a few. There could be a global WildFly / EAP console one day too; then theres the UIs for hawkular's alerts and stuff. 

i.e. there's lots of web apps / consoles. It'd be nice if the OpenShift console made it easy to link to them all; in a kinda 'Start menu' kinda thing like windows used to have.

So it'd be nice if there was a way for a project/namespace to curate a list of apps that the console could link to. Lots of these apps are not necessary 'projects' that folks develop in the usual PaaS sense with a git repo; they are just stuff you run on kubernetes.

e.g. there could be an Consoles tab / link / menu on the UI which lists all the consoles/apps that have been curated for the namespace/project the user is working on. Then a project/team/namespace can be configured to have nice links with icons, descriptions and stuff to link to whatever web apps are running inside a project.

There's already the metadata to generate this list of apps from the kubernetes metadata; but maybe we add extra annotations / labels to populate it automatically. e.g. if folks run kibana and grafana; then they could automatically appear on the Apps/Consoles tab/menu; maybe reusing an icon from an annotation on the service. e.g. maybe we add an annotation for a service to appear on the 'app/console' menu?

Folks might be running 20 services in a namespace but they may only want to show, say, 5 as apps/links on the main openshift console.

I'm happy for the UX folks to come up with however they want to lay this out and style it but imagine something like a new tab in the UI

```
Apps:

* Kibana 
* Grafana
* Jenkins
* Nexus
```

then as folk hover over the links there'd be tooltips and stuff along with icons so folks can easily see what the things are etc

**Comment by smarterclayton** [comment]
Makes sense.  Services should contribute to the console dynamically.

**Comment by smarterclayton** [comment]
The problem right now is external access to the cluster, which is mostly held up by the external IP and service public IP debate.

**Comment by jstrachan** [comment]
Yeah; I'm assuming only services which have a OpenShift Route defined with an external host would be eligible for appearing in the 'apps/consoles' menu. We're using routes right now to good effect in the fabric8 console. 

The implementation could switch to work with public IPs / DNS later on if need be; but at least work today for services folks have created a Route for?

**Comment by liggitt** [comment]
probably something we'd want to handle with an extension

**Comment by smarterclayton** [comment]
The discussion upstream on "components" is relevant.  Also API discovery
docs.

On Sep 10, 2015, at 12:30 PM, Jordan Liggitt <notifications@github.com>
wrote:

probably something we'd want to handle with an extension

—
Reply to this email directly or view it on GitHub
<https://github.com/openshift/origin/issues/1977#issuecomment-139302190>.

**Comment by jstrachan** [comment]
@smarterclayton thanks! Any chance of links?  "components" is a hard term to search for ;). Haven't found any API discovery docs yet either

**Comment by smarterclayton** [comment]
API discovery struct https://github.com/kubernetes/kubernetes/pull/13955

**Comment by jwforres** [comment]
closing this in favor of the UXD R&D trello story https://trello.com/c/iAbEy9X8
